context/app/static/js/components/detailPage/visualization/VisualizationTracker/hooks.ts

```diff
@@ -1,90 +1,102 @@
-import React, { useCallback, useRef } from 'react';
+import React, { useMemo, useRef } from 'react';
 import useTrackMount from 'js/hooks/useTrackMount';
-import { VitessceInteraction } from './types';
-import {
-  getLastInteraction,
-  getNearestIdentifier,
-  handleKeyPress,
-  modifierKeys,
-  mouseButtonMap,
-  pushVitessceEvent,
-} from './utils';
+import { useFlaskDataContext } from 'js/components/Contexts';
+import { useEventCallback } from '@mui/material';
+import { trackEvent } from 'js/helpers/trackers';
+import { getNearestIdentifier, modifierKeys, mouseButtonMap } from './utils';
 
-type FocusAndMouseHandler = React.MouseEventHandler<HTMLSpanElement> & React.FocusEventHandler<HTMLSpanElement>;
+export function useVitessceEventMetadata() {
+  const location = window.location.href;
+  const flaskData = useFlaskDataContext();
 
-export const useVisualizationTracker = () => {
-  // Track when the visualization is first mounted
-  useTrackMount('Visualization', 'Vitessce Mounted');
-  // Create an object to track Vitessce interactions
-  // This list is populated with a summary of every interaction a user has with the visualization
-  const vitessceInteraction = useRef<VitessceInteraction>([]);
+  let category = 'Unknown';
+  let name = location; // Fall back to the full URL if we can't find a better name
+
+  // Handle preview pages
+  if (location.includes('preview')) {
+    const { title } = flaskData;
+    category = 'Preview';
+    name = title;
```

Comment:
I believe we should pass this as `label`. `name` gets set to the value of `label` so it works with both GA/Matomo.

Reply:
Will do!

Reply:
It happens [here](https://github.com/hubmapconsortium/portal-ui/blob/292509d5af96772cef7016e94c5c58d1bc5a060a/context/app/static/js/helpers/trackers.js#L105-L123).

Reply:
Thanks for pointing this out! I switched `name` to `label` where appropriate.